stubs/python-jose/jose/jwt.pyi

```diff
@@ -1,16 +1,25 @@
+from collections.abc import Container, Iterable, Mapping
 from typing import Any
 
-def encode(claims, key, algorithm=..., headers: Any | None = ..., access_token: Any | None = ...): ...
+from .backends.base import Key
+
+def encode(
+    claims: Mapping[str, Any],
+    key: str | dict[str, Any] | Key,
+    algorithm: str = ...,
+    headers: Mapping[str, Any] | None = ...,
+    access_token: str | None = ...,
+) -> str: ...
 def decode(
-    token,
-    key,
-    algorithms: Any | None = ...,
-    options: Any | None = ...,
-    audience: Any | None = ...,
-    issuer: Any | None = ...,
-    subject: Any | None = ...,
-    access_token: Any | None = ...,
-): ...
-def get_unverified_header(token): ...
-def get_unverified_headers(token): ...
-def get_unverified_claims(token): ...
+    token: str,
+    key: str | dict[str, Any] | Key,
```

Comment:
It's passed to the `key` parameter of `jws.verify()`: https://github.com/mpdavis/python-jose/blob/96474ecfb6ad3ce16f41b0814ab5126d58725e2a/jose/jwt.py#L142

```suggestion
    key: str | Mapping[str, Any] | Key,
```